Lib/ast.py

```diff
@@ -613,35 +636,46 @@ def __exit__(self, exc_type, exc_value, traceback):
     def block(self):
         return self._Block(self)
 
-    def traverse(self, node):
+    class _Delimit:
+        """A context manager for preparing the source for expressions. It adds
+        start of the delimiter if condition is true and enters, after exit if the condition
+        was true it adds delimiter end."""
+        def __init__(self, unparser, condition, delimiter):
+            self.unparser = unparser
+            self.delimiter = delimiter
+            self.condition = condition
+
+        def __enter__(self):
+            if self.condition:
+                self.unparser.write(self.delimiter[0])
+
+
+        def __exit__(self, exc_type, exc_value, traceback):
+            if self.condition:
+                self.unparser.write(self.delimiter[-1])
+
+    def delimit(self, condition = True, delimiter = "()"):
+        return self._Delimit(self, condition = condition, delimiter = delimiter)
```

Comment:
Perhaps clearer to drop `condition` here, and use instead:
```
def delimit(self, condition, delimiter):
    if condition:
        return self._Delimit(self, delimiter)
    else:
        return contextlib.nullcontext()
```

Reply:
unfortunately we can't use contextlib due to performance issues.

Reply:
Still tempting to create separate classes, which gives you one branch and not two:
```
class _NoDelimit:
    def __enter__(self): pass
    def __exit__(self, *args): pass
```

Reply:
This is going to make this code more complex and I am not sure if we keep going like this, @pablogsal what do you think about re-bringing contextlib back. Then we can just return `nullcontext` and even better we can just use `contextmanager`